crates/ol/mempool/src/manager.rs

```diff
@@ -0,0 +1,516 @@
+};
+
+/// Default capacity for event broadcast channel.
+const DEFAULT_EVENT_CHANNEL_CAPACITY: usize = 1000;
+
+/// Mempool manager providing synchronous API with database persistence.
+///
+/// This wraps `MempoolCore` with Arc<Mutex<>> and provides methods for:
+/// - RPC handlers to submit transactions
+/// - Block assembly to retrieve transactions
+/// - Management operations (update slot, get stats)
+/// - Database persistence for durability across restarts
+/// - Event broadcasting for transaction state changes
+#[derive(Clone)]
+pub struct MempoolManager {
+    /// Core mempool protected by mutex.
+    core: Arc<Mutex<MempoolCore>>,
+
+    /// Storage for persistence.
+    storage: Arc<NodeStorage>,
+
+    /// Event broadcast sender for notifying subscribers.
+    event_sender: broadcast::Sender<MempoolEvent>,
+}
+
+impl MempoolManager {
+    /// Create a new mempool manager with the given configuration, current slot, and storage.
+    pub fn new(config: MempoolConfig, current_slot: u64, storage: Arc<NodeStorage>) -> Self {
+        let (event_sender, _) = broadcast::channel(DEFAULT_EVENT_CHANNEL_CAPACITY);
+        Self {
+            core: Arc::new(Mutex::new(MempoolCore::new(config, current_slot))),
+            storage,
+            event_sender,
+        }
+    }
+
+    /// Create a new mempool manager and restore transactions from database.
+    ///
+    /// Loads all persisted transactions from the database into the in-memory core.
+    /// Invalid or expired transactions are silently skipped during restoration.
+    pub fn new_with_restore(
+        config: MempoolConfig,
+        current_slot: u64,
+        storage: Arc<NodeStorage>,
+    ) -> MempoolResult<Self> {
+        let manager = Self::new(config, current_slot, storage);
+        manager.restore_from_database()?;
+        Ok(manager)
+    }
+
+    /// Restore transactions from database into in-memory core.
+    fn restore_from_database(&self) -> MempoolResult<()> {
+        let txids = self
+            .storage
+            .mempool()
+            .get_all_tx_ids()
+            .map_err(|e| MempoolError::DatabaseError(e.to_string()))?;
+
+        for txid in txids {
+            // Load transaction from database
+            let tx = self
+                .storage
+                .mempool()
+                .get_tx_entry(&txid)
+                .map_err(|e| MempoolError::DatabaseError(e.to_string()))?;
+
+            if let Some(tx) = tx {
+                // Compute blob size for metadata
+                let blob = encode_to_vec(&tx).map_err(|e| {
+                    MempoolError::ParseError(format!("Failed to encode transaction: {e}"))
+                })?;
+                let blob_size = blob.len();
+
+                // Try to add to core (may fail if invalid/expired - that's ok)
+                let mut core = self.core.lock().unwrap();
+                let _ = core.submit_transaction(tx, blob_size);
+            }
+        }
```

Comment:
Constructing the `MempoolState` (currently `MempoolCore`) should be structured like `open_database(&Database)` call.  Then you can take that state and plug it into the worker task.